include/circt/Dialect/OM/OMOps.td

```diff
@@ -195,4 +195,22 @@ def ListCreateOp : OMOp<"list_create", [Pure, SameTypeOperands]> {
   let hasCustomAssemblyFormat = 1;
 }
 
+def MapCreateOp : OMOp<"map_create", [Pure]> {
+  let summary = "Create a map";
+  let description = [{
+    Creates a map from a sequence of inputs.
+
+    ```
+    %map = om.map_create [%k1 -> %v1, %k2 -> %v2] : !om.string, !om.ref
+    ```
+  }];
+
+  let arguments = (ins Variadic<AnyType>:$inputs);
```

Comment:
Could this instead be variadic of a key-value attribute, constrained to use the key/value types of the map?

Such an attribute/structure would likely make printing the pairs easier (`interleaveComma(os << key << " -> " << value)` or so).

Related: verifier, but perhaps with the right ODS description most of what's useful here would be checked for us.

Reply:
Thanks! Yes, I'll add a verifier. I considered a bit about using variadic key-value pairs as ODS operands (`Variadic<KeyValuePair>: $inputs`) but I guess it will be complicated. I think internal representation of `KeyValuePair` would be `mlir::TupleType` (or maybe `om::TupleType`) that will require tuple/pair support to OM dialect. Another possible implementation would be two variadic operands `(ins Variadic<KeyType>: $keys, Varidiac<AnyType>: $values)` but this approach will require `AttrSizedOperandSegments` trait implementation. Considered that I feel the current representation + verifier + few helpers (`getKeys()` or `getValues()`) might be practically sufficient and simple. WDYT?

Reply:
One thing to consider: we may need Tuple support in OM anyway. For "maps" with non-string key, we may need to represent a List<Tuple>.

Reply:
Sounds good, @uenoku -- I think you're right that would be moderately more difficult to make work neatly, thank you for the answer/explanation.  It was meant to be a way to make it simpler!

That said, if we need Tuple anyway then maybe that's the way to go, but up to you.   Thanks!

Reply:
Thank you for the suggestion. I added tuple operations in https://github.com/llvm/circt/pull/5878 and changed create_map's operand type to `tuple<om.string, value`.